rasterio/features.py

```diff
@@ -483,3 +491,149 @@ def is_valid_geom(geom):
+        Include nodata regions?
+
+    geographic: str (DEFAULT: True)
+        Output shapes in EPSG:4326? Otherwise use the native CRS.
+
+    precision: int (DEFAULT: -1)
+        Decimal precision of coordinates. -1 for full float precision output
+
+    Yields
+    ------
+    GeoJSON-like Feature dictionaries for shapes found in the given band
+    """
+    if bidx is not None and bidx > src.count:
+        raise ValueError('bidx is out of range for raster')
+
+    img = None
+    msk = None
+
+    # Adjust transforms.
+    transform = src.transform
+    if sampling > 1:
+        # Determine the target shape (to decimate)
+        shape = (int(math.ceil(src.height / sampling)),
+                 int(math.ceil(src.width / sampling)))
+
+        # Calculate independent sampling factors
+        x_sampling = src.width / shape[1]
+        y_sampling = src.height / shape[0]
+
+        # Decimation of the raster produces a georeferencing
+        # shift that we correct with a translation.
+        transform *= Affine.translation(
+            src.width % x_sampling, src.height % y_sampling)
+
+        # And follow by scaling.
+        transform *= Affine.scale(x_sampling, y_sampling)
+
+    # Most of the time, we'll use the valid data mask.
+    # We skip reading it if we're extracting every possible
+    # feature (even invalid data features) from a band.
+    if not band or (band and not as_mask and not with_nodata):
+        if sampling == 1:
+            msk = src.read_masks(bidx)
+        else:
+            msk_shape = shape
+            if bidx is None:
+                msk = np.zeros(
+                    (src.count,) + msk_shape, 'uint8')
+            else:
+                msk = np.zeros(msk_shape, 'uint8')
+            msk = src.read_masks(bidx, msk)
+
+        if bidx is None:
+            msk = np.logical_or.reduce(msk).astype('uint8')
+
+        # Possibly overridden below.
+        img = msk
+
+    # Read the band data unless the --mask option is given.
+    if band:
+        if sampling == 1:
+            img = src.read(bidx, masked=False)
+        else:
+            img = np.zeros(
+                shape,
+                dtype=src.dtypes[src.indexes.index(bidx)])
+            img = src.read(bidx, img, masked=False)
+
+    # If as_mask option was given, convert the image
+    # to a binary image. This reduces the number of shape
+    # categories to 2 and likely reduces the number of
+    # shapes.
+    if as_mask:
+        tmp = np.ones_like(img, 'uint8') * 255
+        tmp[img == 0] = 0
+        img = tmp
+        if not with_nodata:
+            msk = tmp
+
+    # Transform the raster bounds.
+    bounds = src.bounds
+    xs = [bounds[0], bounds[2]]
+    ys = [bounds[1], bounds[3]]
+    if geographic:
+        xs, ys = warp.transform(
+            src.crs, CRS({'init': 'epsg:4326'}), xs, ys)
+    if precision >= 0:
+        xs = [round(v, precision) for v in xs]
+        ys = [round(v, precision) for v in ys]
```

Comment:
@perrygeo I don't know how I missed it before, but I think `xs` and `ys` are never used and therefore lines 605-614 can be ✂️